cve_bin_tool/cvedb.py

```diff
@@ -115,6 +115,13 @@ class CVEDB:
             PRIMARY KEY(metrics_id)
         )
         """,
+        "deduplication": """
+        CREATE TABLE IF NOT EXISTS deduplication (
+            purl TEXT,
+            vendor TEXT,
```

Comment:
Hm, in my head this would have been CPEs or vendor, product instead of just vendor.    

Is there a case where we might want one entry from a (very large) vendor but not another?  I'm thinking stuff like say we're looking for "commons" and it's actually `apache:commons-io` that we want but a search for commons would also turn up other commons projects like `apache:commons-httpclient`.  I think we have an open bug with some examples you could play with to see how they work right now.

How hard would it be to add a product field here if it turns out we need it later?

Reply:
Well, product info is already in the purl so I thought why make it redundant. I think we _can_ add CPEs, but _should_ we? Don't we only need vendor info?

We'll mostly (maybe always) search regarding a product with complete product names (like commons-io). But I'll look into the bug and see if we need to modify anything.

Adding product field will take only a line of change, as we don't really have much data for deduplication.

Reply:
Yeah, let's leave it as is for now.  But we did have an issue filed recently that mentioned that java in particular has a lot of libraries named "commons" and the way we're reading maven files passes them through without adding stuff like `-io` so we may hit this sooner rather than later.